webapp/resources/templates/fragments/loginProviders.html

```diff
@@ -22,7 +22,7 @@ <h3 th:utext="#{screen.welcome.label.loginwith}"/>
                     <span th:case="linkedin" class="fa fa-linkedin"></span>
                     <span th:case="facebook" class="fa fa-facebook-square"></span>
                     <span th:case="oidc" class="fa fa-openid"></span>
-                    <span th:case="saml2" class="fa fa-lock"></span>
+                    <span th:case="saml2" th:class="${entry.cssClass}" ></span>
```

Comment:
Is this really something that needs to be calculated in the java code? Cant we calculate the same thing here in the UX, given the thing you're working on is related to UX anyway?